src/processors/universal.ts

```diff
@@ -471,45 +539,81 @@ class UniversalProcessor extends BaseOrderProcessor {
+            evmFulfillGasLimit,
+            evmFulfillCappedGasPrice: evmFulfillCappedGasPrice ? BigInt(evmFulfillCappedGasPrice.integerValue().toString()) : undefined,
+            swapEstimationPreference: preswapTx
+          }
+      );
+    } catch (e) {
+      const error = e as Error;
+      this.hooksEngine.handleOrderPostponed({
+        order: orderInfo,
+        estimation: undefined,
+        context,
+        reason: PostponingReason.ESTIMATION_FAILED,
+        message: error.message,
+      });
+      context.logger.error(`Error in estimation ${e}`);
+      context.logger.error(e);
+      return;
+    }
+
     const {
       reserveDstToken,
       requiredReserveDstAmount,
       isProfitable,
       reserveToTakeSlippageBps,
-    } = await calculateExpectedTakeAmount(
-      orderInfo.order,
-      this.params.minProfitabilityBps,
-      {
-        client: context.config.client,
-        giveConnection: context.giveChain.fulfillProvider.connection as Web3,
-        takeConnection: this.takeChain.fulfillProvider.connection as Web3,
-        priceTokenService: context.config.tokenPriceService,
-        buckets: context.config.buckets,
-        swapConnector: context.config.swapConnector,
-        logger: createClientLogger(logger),
-        batchSize,
-        evmFulfillGasLimit,
-        evmFulfillCappedGasPrice: evmFulfillCappedGasPrice ? BigInt(evmFulfillCappedGasPrice.integerValue().toString()) : undefined,
-        swapEstimationPreference: preswapTx
-      }
-    );
+    } = estimation;
 
+    const hookEstimation = {
+      isProfitable,
+      reserveToken: reserveDstToken,
+      requiredReserveAmount: requiredReserveDstAmount,
+      fulfillToken: orderInfo.order?.take.tokenAddress!,
+      projectedFulfillAmount: orderInfo.order.take.amount!.toString(),
+    };
+    this.hooksEngine.handleOrderEstimated({
+      order: orderInfo,
+      estimation: hookEstimation,
+      context,
+    });
 
     if (isProfitable) {
-      logger.info("order is profitable")
+      logger.info("order is profitable");
     }
     else {
       logger.info("order is not profitable");
+      this.hooksEngine.handleOrderPostponed({
+        order: orderInfo,
+        estimation: hookEstimation,
+        context,
+        reason: PostponingReason.NON_PROFITABLE,
+      });
       if (allowPlaceToMempool)
         this.mempoolService.addOrder(params);
       return;
     }
 
     if (!buffersAreEqual(reserveDstToken, pickedBucket.reserveDstToken)) {
-      logger.error(`internal error: \
+      const message = `internal error: \
 dln-taker has picked ${tokenAddressToString(orderInfo.order.take.chainId, pickedBucket.reserveDstToken)} as reserve token, \
-while calculateExpectedTakeAmount returned ${tokenAddressToString(orderInfo.order.take.chainId, reserveDstToken)}`);
-      return;
+while calculateExpectedTakeAmount returned ${tokenAddressToString(orderInfo.order.take.chainId, reserveDstToken)}`;
+      logger.error(message);
```

Comment:
Should we use logger.error here? We already catch and print all errors in `tryProcess()`